### DISCUSSION TITLE
After Flashing ESP32, No "Tasmota_XXXX" SSID...How To Connect?

### DISCUSSION THREAD

**Discussion by TSJim** [arendst/Tasmota#10340] [Q&A] [answered]
Hey, I need a little help figuring out what I'm doing wrong.

Tonight I flashed the "tasmota32.bin" file to a Wemos LOLIN32 dev board.  Everything seemed to work right, and I got the "Hash of data verified" message.  Then rebooted it.

I expected to see an SSID like "Tasmota-XXXX" on WiFi, similar to what happens when you flash Tasmota to a Tuya switch or plug, that I would then connect to and use the WiFiManager functionality to put it on my WiFi network, but I'm not seeing anything.

Is there a different way that I need to connect to a newly Tasmota flashed ESP32?

**Comment by barbudor** [comment, marked as answer]
Hi @TSJim
Happy new year

Did you also flashed the 3 other bins (bootloader, partitions, and the 3rd I can't remember)?
All 3 files are mandatory beside the tasmota app itself. 
Please refer to the docs and files in Tasmota/tools/Esptool/ESP32/

Cheers

**Comment by TSJim** [comment]
Hey Barbudor...

Yep, that was the problem.

Here is the command line that worked for me (after booting the ESP32 dev board with GPIO0 grounded):

`sudo esptool.py --port /dev/cu.SLAB_USBtoUART write_flash 0x1000 /Users/jim/Downloads/BootLoader.bin 0x8000 /Users/jim/Downloads/Partition_Table.bin 0x10000 /Users/jim/Downloads/tasmota32.bin 0x225000 /Users/jim/Downloads/SPIFFS.bin`

I was able to write the files successfully and get the ESP32 connected to Home Assistant!

Thank you.

**Reply by barbudor** [reply]
Are you using the files provided by Tasmota ?
The names are different
The command line is also different from what is provided in the Tasmota doc or the README

![image](https://user-images.githubusercontent.com/2996042/103446415-9a680d00-4c7f-11eb-9e58-615dd4e1c5b1.png)

```
esptool.py --chip esp32 --port COM5 --baud 921600 --before default_reset --after hard_reset write_flash -z --flash_mode dout --flash_freq 40m --flash_size detect 0x1000 bootloader_dout_40m.bin 0x8000 partitions.bin 0xe000 boot_app0.bin 0x10000 tasmota32.bin
```

**Comment by ovisopa** [comment]
Anyone knows if this is related to my problem ? I had a brand new ESP32 D1 mini and flashed tasmota32.factory.bin, tasmota32.bin don't really know in what order, the thing is it never connected to wifi. I'm not used to tasmota, I don't understand how to flash 3 different binf files without deleting what other file installed. After seeing I can't make it work I installed other firmwares too, like tasmota32solo ... nothing works. 

in the install web if I open the log it always shows this , the AP name is correct. Shouldn't the tasmota create it's own AP if it can't connect to the saved wifi so I can change the wifi and password (I'm 100% percent sure the ap and password are correct, as Chrome browser autocompleted the info from my saved passwords)

<img width="1041" height="457" alt="image" src="https://github.com/user-attachments/assets/123b9782-6725-43a1-93ae-25f79e1439d4" />

Any help is appreciated.

**Comment by ovisopa** [comment]
Can someone send me the links where I can find the other 2 bin files that I need to flash besides tasmota32.factory.bin  ?

I tried flashing factory.bin from web installer , also from esptool on windows but the board never showed any Tasmota-xxx AP after reboot, When in console I run Wifiscan 1... it never finds any network, even there are many neighbors here, also GPT suggests to flash the other 2 bins, but it can provide any links to those bin files for my board

```
python -m esptool --chip esp32 --port COM7 write_flash -z \
  0x1000 bootloader_dio_80m.bin \
  0x8000 partitions.bin \
  0x10000 tasmota32.factory.bin
```

My Board details MINI D1 ESP32

`Flash memory erased successfully in 7.6 seconds. 
Hard resetting via RTS pin... 
PS C:\Users\Lenovo IdeaPad\Downloads> python -m esptool --chip esp32 --port COM7 write_flash -z 0x1000 tasmota32.factory.bin Warning: Deprecated: Command 'write_flash' is deprecated. Use 'write-flash' instead. esptool v5.0.2 
Connected to ESP32 on COM7: 
Chip type: ESP32-D0WD-V3 (revision v3.1) 
Features: Wi-Fi, BT, Dual Core + LP Core, 240MHz, Vref calibration in eFuse, Coding Scheme None 
Crystal frequency: 40MHz 
MAC: ec:e3:34:7b:8b:84 
Stub flasher running. 
Configuring flash size... Flash will be erased from 0x00001000 to 0x002e0fff... Wrote 3011760 bytes (1869154 compressed) at 0x00001000 in 171.2 seconds (140.7 kbit/s). Hash of data verified. 
Hard resetting via RTS pin... 
PS C:\Users\Lenovo IdeaPad\Downloads>`

Thank you

**Reply by sfromis** [reply]
The complicated procedure with flashing 3 different files is not used now, as you can just flash `tassmota.factory.bin` (which contains bootloader and partitions too) at location 0x000000. The binary `tasmota32.bin` is not to be used for initial flashing, but only for OTA upgrade when Tasmota is already flashed.

You don't even need to construct commands for esptool.py, as you can use the Tasmota Web Installer without even downloading anything first. It will also automatically select the right file, you only have to select what port you use. https://tasmota.github.io/install/ (which it looks like you were using)

And if you by "GPT" mean asking ChatGPT for help, that was a bad plan, as you often get directly misleading responses for narrow topics.

When it comes to getting connected, the screenshot you posted looks like Tasmota is trying to connect (you must have connected to the Tasmota AP already, to enter network name and password). Symptoms looks like bad password. It is not a given that any password you've used will work, as the recommendation is to use password without special characters. Another risk factor is if your USB port cannot supply sufficient power to get on Wifi.

**Reply by ovisopa** [reply]
> you must have connected to the Tasmota AP already

After installing firmware I press NEXT and reached CONNECT TO WIFI step, no networks ware found, but I have my wifi SSID and password saved in my browser autocomplete, so I did complete the "manually" from google chrome, not from an dropdown box with all the networks my board found (it should find neighbors too, but it did not find any)

This is why it tries to connect to an existing network, because I wrote the ssid and password myself, not because the board find it

After many reflashing using the web installer I reached GPT and did install esptool to try that way too, than he sent me some suggestions to use the console to push a new wifi ssid to the board
`Backlog SSID1 testwifi; Password1 12345678; Restart 1`
tried to make an hotspot on my phone, I got the same result, it never could connect to it

By using 
`WifiScan 1
`
it never find any networks, it should find neighbors wifi's, not just mine.

00:02:26.463 CMD: WifiScan 1 
00:02:26.465 RSL: RESULT = {"WiFiScan":"Scanning"} 
00:02:29.462 RSL: RESULT = {"WiFiScan":"No networks found"} 
00:02:36.461 WIF: Connect failed with AP timeout

**Reply by ovisopa** [reply]
> Symptoms looks like bad password. It is not a given that any password you've used will work, as the recommendation is to use password without special characters

The password was saved in the browser autocomplete, I have used it in the other 6 ESP8266/ESP32 devices I have in that location. And why it would not find any network when sending to console Wifiscan 1 ? 

> Another risk factor is if your USB port cannot supply sufficient power to get on Wifi.

I did flash other ESP devices using that laptop, and it worked, I don't think this can be the problem
